### PR TITLE
add class

### PR DESCRIPTION
**Pull request by Seragors**

### CODE REVIEW

File: src/main/java/Main.java [hunk 1]
```diff
@@ -1,5 +1,50 @@
...
+import io.lettuce.core.RedisClient;
+import org.hibernate.Session;
+import redis.RedisService;
+
+import java.util.List;
+
+import static java.util.Objects.nonNull;
+
 public class Main {
+    static final SessionDataBase CONNECTION_DATABASE = new SessionDataBase();
```
Comment (Volodymyr95) [inline src/main/java/Main.java]: make separate class like Menu and move all this code here

File: src/main/java/Main.java [hunk 2]
```diff
@@ -1,5 +1,50 @@
...
+    static final Session currentSession = CONNECTION_DATABASE.getCurrentSession();
+    static final CityDAO cityDAO = new CityDAO(City.class, currentSession.getSessionFactory());
+    static final ObjectMapper mapper = new ObjectMapper();
+    static final ClientRedisDataBase clientRedisDataBase = new ClientRedisDataBase();
+    static final RedisClient redisClient = clientRedisDataBase.prepareRedisClient();
+    static final RedisService transformator = new RedisService();
+
     public static void main(String[] args) {
-        System.out.println("Hello world!");
+        List<City> all = cityDAO.findAll();
```
Comment (Volodymyr95) [inline src/main/java/Main.java]: the same

File: src/main/java/connect/ClientRedisDataBase.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+public class ClientRedisDataBase {
+    public RedisClient prepareRedisClient() {
+        RedisClient redisClient = RedisClient.create(RedisURI.create("localhost", 6379));
+        try (StatefulRedisConnection<String, String> connection = redisClient.connect()) {
+            System.out.println("\nConnected to Redis\n");
+        }
+        return redisClient;
+    }
+
+    public void testRedisData(List<Integer> ids, RedisClient redisClient, ObjectMapper mapper) {
```
Comment (Volodymyr95) [inline src/main/java/connect/ClientRedisDataBase.java]: why u added this method?

File: src/main/java/connect/ClientRedisDataBase.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.lettuce.core.RedisClient;
+import io.lettuce.core.RedisURI;
+import io.lettuce.core.api.StatefulRedisConnection;
+import io.lettuce.core.api.sync.RedisStringCommands;
+import redis.CityCountry;
+
+import java.util.List;
+
+public class ClientRedisDataBase {
```
Comment (Volodymyr95) [inline src/main/java/connect/ClientRedisDataBase.java]: RedisUtil

File: src/main/java/connect/ClientRedisDataBase.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    public RedisClient prepareRedisClient() {
```
Comment (Volodymyr95) [inline src/main/java/connect/ClientRedisDataBase.java]: getRedisClient

File: src/main/java/connect/SessionDataBase.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.cfg.Configuration;
+import org.hibernate.cfg.Environment;
+
+import java.util.List;
+import java.util.Properties;
+import java.util.Set;
+
+public class SessionDataBase {
```
Comment (Volodymyr95) [inline src/main/java/connect/SessionDataBase.java]: HibenrnateUtil

File: src/main/java/connect/SessionDataBase.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+        sessionFactory = new Configuration()
+                .addAnnotatedClass(City.class)
+                .addAnnotatedClass(Country.class)
+                .addAnnotatedClass(CountryLanguage.class)
+                .setProperties(properties)
+                .buildSessionFactory();
+    }
+
+    public void testMySQLData(List<Integer> ids, CityDAO cityDAO) {
```
Comment (Volodymyr95) [inline src/main/java/connect/SessionDataBase.java]: delete this

File: src/main/java/dao/EntityDAO.java [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+        transaction.commit();
+        return singleResult;
+    }
+
+    public List<T> findAll() {
+        List<T> resultList = new ArrayList<>();
+        int step = 500;
+        int totalCount = getTotalCount().intValue();
+        Transaction transaction = getCurrentSession().beginTransaction();
+        for (int i = 0; i < totalCount; i += step) {
```
Comment (Volodymyr95) [inline src/main/java/dao/EntityDAO.java]: make it using query

File: src/main/java/connect/ClientRedisDataBase.java (hunk 3 above)
Comment (Seragors) [reply src/main/java/connect/ClientRedisDataBase.java]: Data from a redis client accepting a list of IDs to retrieve

File: src/main/java/Main.java [hunk 9]
```diff
@@ -1,5 +1,5 @@
 public class Main {
     public static void main(String[] args) {
-        System.out.println("Hello world!");
+        new Menu().open();
```
Comment (Volodymyr95) [inline src/main/java/Main.java]: new Menu().run()

File: src/main/java/Menu.java [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+    static final RedisClient redisClient = REDIS_UTIL.getRedisClient();
+    static final RedisService transformator = new RedisService();
+    Scanner scanner = new Scanner(System.in);
+
+    public void open() {
+        while (isRun) {
+            System.out.println(MENU_SELECT);
+            int number = scanner.nextInt();
+            if (number == ALL_CITY) {
+                System.out.println(ALL_CITY);
```
Comment (Volodymyr95) [inline src/main/java/Menu.java:34]: move this logic to services

**Review by Volodymyr95 — APPROVED** [review]

**Pull request merged**